### ISSUE TITLE
Zone Support

### ISSUE THREAD

**Issue by gavc1** [gondalez/homebridge-izone-airconditioner#3]
The system controls are exposed to HomeKit, but the zones are not. 
iZone has the ability to have multiple zones with independent temperatures and target temperatures, would it be possible to add each zone as an accessory to HomeKit?

**Comment by gondalez** [comment]
Hi @gavc1,

Yes I think it is possible but a bit beyond my spare time capacity at the moment :) 

I have been considering this as a kind of road map for zones so it can be done in chunks:

- add a thermometer to show the current temperature for all enabled zones (handy on its own)
- add a switch to open/close those zones without climate control
- add a switch to open/close/climate-control those zones with climate control
- add a thermostat control for those zones with climate control (hide when the zone is set to open or closed)

Unanswered questions

- how to determine which zones have climate control available? The api has [`NoOfZones`](https://github.com/gondalez/homebridge-izone-airconditioner/blob/45db49d379c9b92d620d7c899ab4df24dc959668/src/__fixtures__/systemSettings.js#L8) which is handy, but I don't see a way to determine the zones with climate control. Worst case I think this can be detected by seeing a non-zero value for [`Temp`](https://github.com/gondalez/homebridge-izone-airconditioner/blob/45db49d379c9b92d620d7c899ab4df24dc959668/src/__fixtures__/zones.js#L10) for the zone.
- does homekit support custom selector values in combination with thermostat controls? What's the best way to implement this.
- how to implement the hiding/showing/disabling of the zone thermostat based on the zone open/closed/climate-control setting

Thoughts welcome :) 
Also, are you using this plugin at home?

**Comment by nickbower** [comment]
Hi @gondalez perhaps points 2-4 above could be simplified if someone ever does this.

For zones without climate control, you want to be able to open/close and additionally be able to control airflow (10 to 100).  I'm in this category and it sounded to me when talking to installer that very few people have all zones with climate control.  It's more likely there is a particular room that gets alot of sun and people are trying to dynamically adjust airflow for that special case.

For those zones with climate control, you just want to open/close them knowing the system will manage airflow.  Trying to set temperature per zone sounds like it's possible according to @gavc1, but I wonder how many people actually operate their aircon/heating like that.  Particularly with the cost of the wireless wall plate sensors (which even themselves only permit local open/close).

**Comment by gondalez** [comment]
Thanks for the thoughtful message @nickbower 🙏 

The optional climate control sensors is the part that has put me off taking on this work to be honest 😅 

I find the main things I want are to turn the whole system on/off and also have the average temperature across zones where climate control is present.
Things like airflow, climate control temperature are not something you'd adjust that often and so I'm ok with doing it on the controller panel.
Also there is the awkward part of what does it mean to adjust a zone via homekit when the entire system is off? Should it turn on the system? 🤔 

BTW I personally bought the battery powered climate control panels for my system as it saves power by keeping the temperature more consistent in the house. Previously I would have a really cold area (which uses more power) as the only sensor was in the return vent.

**Comment by NickBower-Woodside** [comment]
yeah sounds similar.  Did you change one zone to a constant so you could install these sensors?

You're right it's fiddly.  There's the other issue of what protection logic would be required to enable a user to accidentally turn off all zones and block airflow.  I've never tested but wonderd if that exists within the panel or the controller unit, and what that would mean for a homekit plugin if it was ever attempted.

**Comment by nickbower** [comment]
yeah sounds similar. Did you change one zone to a constant so you could install these sensors?

You're right it's fiddly. There's the other issue of what protection logic would be required to enable a user to accidentally turn off all zones and block airflow. I've never tested but wonderd if that exists within the panel or the controller unit, and what that would mean for a homekit plugin if it was ever attempted.

**Comment by gondalez** [comment]
Oh yes good point around airflow / all-off.

Now that you mention it I did set a single constant as I read somewhere online that not setting one is bad for the unit.
I wasn't game to try no constants it lest it blew something up 😅 

Another thing that may be useful to know is that iZone have come out with a controller that has official homekit support. 
I found this just now: https://www.izonepedia.com.au/index.htm?context=2340

I haven't read it in depth yet but it might help to see how they implemented zones.
